pbs_light_test.go

```diff
@@ -623,3 +625,28 @@ func (validator *testValidator) Schema(name openrtb_ext.BidderName) string {
 		return "{\"appnexus\":false}"
 	}
 }
+
+// Test the viper setup
+func TestViperInit(t *testing.T) {
+	CompareStrings(t, "Viper error: external_url expected to be %s, found %s", "http://localhost:8000", viper.Get("external_url").(string))
+	CompareStrings(t, "Viper error: adapters.pulsepoint.endpoint expected to be %s, found %s", "http://bid.contextweb.com/header/s/ortb/prebid-s2s", viper.Get("adapters.pulsepoint.endpoint").(string))
+}
+
+func TestViperEnv(t *testing.T) {
+	os.Setenv("PBS_PORT", "7777")
```

Comment:
hmm..... is there any way to test this _without_ mutating state inside the test? As a general rule, tests which mutate global state are bad news.

Maybe better to set these `ENV` vars inside `./validate.sh`? Or at the very least, use some `defer` statements to reset these to their original values when the test is over.

Also: `os.Setenv` returns an error... it's probably worth checking that and `t.Fatal`-ing if the set didn't work. It's likely to be easier to debug than if `CompareStrings` fails with no other explanation.

Reply:
On second thought..... meh. I just checked Viper's tests, and they just call `os.Setenv` as well: https://github.com/spf13/viper/blob/master/viper_test.go#L364

It is probably still worthwhile to check the error returns though.

Reply:
Yes, I had an eye on the viper tests when I wrote mine. Still probably not a bad idea to unset at the end.